File: Program/LawCreator/src/pl/wroc/pwr/student/lawcreator/xml/ElementImpl.java
/*
 * LawCreator jest narzedziem, usprawniajacym pisanie aktów 
 * prawnych zgodnie ze schemtem narzuconym przez MSWiA. System 
 * umozliwia uzytkownikowi tworzenie jednego z dokumentow 
 * umieszczonych w pliku schema.xsd, jakim jest uchwala. Serwis 
 * umozliwia wygenerowanie dokumentu XML.
 * 
 * Copyright (C) 2012 Adamiak Mateusz
 * Niniejszy program jest wolnym oprogramowaniem; mo¿esz go 
 * rozprowadzaæ dalej i/lub modyfikowaæ na warunkach Powszechnej
 * Licencji Publicznej GNU, wydanej przez Fundacjê Wolnego
 * Oprogramowania - wed³ug wersji 2-giej tej Licencji lub którejœ
 * z póŸniejszych wersji.
 * 
 * Niniejszy program rozpowszechniany jest z nadziej¹, i¿ bêdzie on
 * u¿yteczny - jednak BEZ JAKIEJKOLWIEK GWARANCJI, nawet domyœlnej
 * gwarancji PRZYDATNOŒCI HANDLOWEJ albo PRZYDATNOŒCI DO OKREŒLONYCH
 * ZASTOSOWAÑ. W celu uzyskania bli¿szych informacji - Powszechna
 * Licencja Publiczna GNU.
 * 
 * Z pewnoœci¹ wraz z niniejszym programem otrzyma³eœ te¿ egzemplarz
 * Powszechnej Licencji Publicznej GNU (GNU General Public License);
 * Jeœli nie - napisz do Free Software Foundation, Inc., 675 Mass Ave,
 * Cambridge, MA 02139, USA.
 */
package pl.wroc.pwr.student.lawcreator.xml;

import java.util.ArrayList;
import java.util.List;

/**
 * Przechowuje i udostepnia dane o elemencie dokumentu XML.
 * 
 * @author Mateusz Adamiak
 * 
 */
public class ElementImpl implements Element {
	private String name = "";
	private String type = "";
	private String description = "";
	private String minOccurs = "1";
	private String maxOccurs = "1";
	private List children;
	private List attributes;

	/**
	 * Tworzy obiekt elementu XML.
	 */
	public ElementImpl() {
		children = new ArrayList();
		attributes = new ArrayList();
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public void setName(String name) {
		this.name = name;
	}

	@Override
	public String getDescription() {
		return description;
	}

	@Override
	public void setDescription(String description) {
		this.description = description;
	}

	@Override
	public String getType() {
		return type;
	}

	@Override
	public void setType(String type) {
		this.type = type;
	}

	@Override
	public String getMinOccurs() {
		return minOccurs;
	}

	@Override
	public void setMinOccurs(String minOccurs) {
		this.minOccurs = minOccurs;
	}

	@Override
	public String getMaxOccurs() {
		return maxOccurs;
	}

	@Override
	public void setMaxOccurs(String maxOccurs) {
		this.maxOccurs = maxOccurs;
	}

	@Override
	public List getElements() {
		return children;
	}

	@Override
	public void add(Element child) {
		children.add(child);
	}

	@Override
	public List getAttributes() {
		return attributes;
	}

	@Override
	public void addAttribute(Attribute attribute) {
		attributes.add(attribute);
	}

	@Override
	public Attribute getAttribute(String name) {
		for (Object each : attributes) {
			if (((Attribute) each).getName().equals(name)) {
				return (Attribute) each;
			}
		}
		return null;
	}

}

File: Program/LawCreator/bin/pl/wroc/pwr/student/lawcreator/xml/ElementImpl.js
Clazz.declarePackage ("pl.wroc.pwr.student.lawcreator.xml");
Clazz.load (["pl.wroc.pwr.student.lawcreator.xml.Element"], "pl.wroc.pwr.student.lawcreator.xml.ElementImpl", ["java.util.ArrayList"], function () {
c$ = Clazz.decorateAsClass (function () {
this.name = "";
this.type = "";
this.description = "";
this.minOccurs = "1";
this.maxOccurs = "1";
this.children = null;
this.attributes = null;
Clazz.instantialize (this, arguments);
}, pl.wroc.pwr.student.lawcreator.xml, "ElementImpl", null, pl.wroc.pwr.student.lawcreator.xml.Element);
Clazz.makeConstructor (c$, 
function () {
this.children =  new java.util.ArrayList ();
this.attributes =  new java.util.ArrayList ();
});
Clazz.overrideMethod (c$, "getName", 
function () {
return this.name;
});
Clazz.overrideMethod (c$, "setName", 
function (name) {
this.name = name;
}, "~S");
Clazz.overrideMethod (c$, "getDescription", 
function () {
return this.description;
});
Clazz.overrideMethod (c$, "setDescription", 
function (description) {
this.description = description;
}, "~S");
Clazz.overrideMethod (c$, "getType", 
function () {
return this.type;
});
Clazz.overrideMethod (c$, "setType", 
function (type) {
this.type = type;
}, "~S");
Clazz.overrideMethod (c$, "getMinOccurs", 
function () {
return this.minOccurs;
});
Clazz.overrideMethod (c$, "setMinOccurs", 
function (minOccurs) {
this.minOccurs = minOccurs;
}, "~S");
Clazz.overrideMethod (c$, "getMaxOccurs", 
function () {
return this.maxOccurs;
});
Clazz.overrideMethod (c$, "setMaxOccurs", 
function (maxOccurs) {
this.maxOccurs = maxOccurs;
}, "~S");
Clazz.overrideMethod (c$, "getElements", 
function () {
return this.children;
});
Clazz.overrideMethod (c$, "add", 
function (child) {
this.children.add (child);
}, "pl.wroc.pwr.student.lawcreator.xml.Element");
Clazz.overrideMethod (c$, "getAttributes", 
function () {
return this.attributes;
});
Clazz.overrideMethod (c$, "addAttribute", 
function (attribute) {
this.attributes.add (attribute);
}, "pl.wroc.pwr.student.lawcreator.xml.Attribute");
Clazz.overrideMethod (c$, "getAttribute", 
function (name) {
for (var each, $each = this.attributes.iterator (); $each.hasNext () && ((each = $each.next ()) || true);) {
if ((each).getName ().equals (name)) {
return each;
}}
return null;
}, "~S");
});
